index.js

```diff
@@ -61,23 +62,34 @@ const RULE_TYPES = {
     const result = await lilconfig("dev-doctor", {
       ignoreEmptySearchPlaces: false,
     }).search();
-    spinner.info("The doctor is in! Performing checkup. . .");
+    spinner.info(
+      `${chalk.underline.bold("The doctor is in")}! Performing checkup. . .`
+    );
 
     // now check repo for each config item to see if it generates error
-    result.config.forEach((rule) => {
-      const { type, ...opts } = rule;
+    for (var i = 0, len = result.config.length; i < len; i++) {
+      const { type, ...opts } = result.config[i],
+        rule = result.config[i];
 
       // Grab metadata for the standard rules
       const enhancedRule = Object.assign({}, rule, getMeta(type));
 
       spinner.start(enhancedRule.description);
 
       if (rules[type]) {
-        const ruleResult = { ...rule, pass: rules[type](opts) };
-        results.push(ruleResult);
-        spinner[ruleResult.pass ? "succeed" : "fail"]();
+        try {
+          const ruleResult = { ...rule, pass: await rules[type](opts) };
```

Comment:
if the rule returns a promise, when we can handle it (and if it doesn't non-promise stuff is treated like `Promise.resolve()`